tools/level_scripts/entities.lua

```diff
@@ -127,6 +127,7 @@ for _, fizzler in pairs(sk_scene.nodes_for_type('@fizzler')) do
         1,
         1,
         room_index,
+        signals.signal_index_for_name(fizzler.arguments[1] or ''),
```

Comment:
If no signal name is specified, this will cause a signal to be defined with an empty name. You need the pattern used by other objects with optional signals which is to use an index of `-1` if the name is empty.

Also, since objects in .blend files need unique names, in maps with 2 chambers (such as `test_chamber_00` and `test_chamber_07`), fizzler objects are named like `@fizzler`, `@fizzler 001`, etc. With your changes, extra unused signals are created, since the unique ID is in the same position as a signal name would be.

To handle this, in addition to the empty check, you can follow the same approach as `@trigger`s and use an index of `-1` if the signal name is equal to `-1`, then edit fizzlers in relevant maps to include a `-1` signal name before their unique ID.

Lastly, a check is needed in `fizzler.c` for whether there is a signal for the fizzler.

I still agree with this change, but this kind of thing is why I'd like to keep level object definition modifications to a minimum for now.

Reply:
Thanks for the in-depth feedback. I did notice that unnecessary empty-named signals were added previously. But I didn't think much of it, since it was the same for buttons. Also since signals are implemented very efficiently concerning memory and there aren't too many fizzler objects in the game. I still implemented your suggested change, as this is clearly more sanitary.

To avoid code duplication, I removed the `parse_trigger_signal` function in `trigger.lua` and moved it to `signals.lua`, renamed as `optional_signal_index_for_name`. This is now used for `@trigger` objects and `@fizzler` objects. I didn't investigate why this is handled differntly for buttons (not all buttons use the cubeSignal after all), but I'd suspect we could handle this the same way.

> I still agree with this change, but this kind of impact is why I'd like to keep breaking level object definition modifications to a minimum for now.

I get what you mean. But given the rather small number of objects in this game, I personally don't think renaming a handful of things in blender is too much of a hassle, really. It is of course a bit more annoying to handle with git, since blend files are binaries...

Reply:
Yeah I was thinking about code cleanliness rather than efficiency. Good to leave things in a nice state and avoid adding more than necessary. Thanks.

As for breaking level object name changes, you're right that renaming a few things isn't the end of the world. I mean that before doing too much here it's best to understand more of the types of changes that may be needed so we can do it in fewer passes.